airbyte-integrations/connectors/source-close-com/source_close_com/source.py

```diff
@@ -0,0 +1,479 @@
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        params["_type"] = self._type
+        if stream_state.get(self.cursor_field):
+            params["date_created__gt"] = stream_state.get(self.cursor_field)
+        return params
+
+
+class OpportunityDueTasks(IncrementalCloseComStream):
+    cursor_field = "date_created"
+    _type = "opportunity_due"
+
+    def path(self, **kwargs) -> str:
+        return "task"
+
+    def request_params(self, stream_state=None, **kwargs):
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        params["_type"] = self._type
+        if stream_state.get(self.cursor_field):
+            params["date_created__gt"] = stream_state.get(self.cursor_field)
+        return params
+
+
+class IncomingSmsTasks(IncrementalCloseComStream):
+    cursor_field = "date_created"
+    _type = "incoming_sms"
+
+    def path(self, **kwargs) -> str:
+        return "task"
+
+    def request_params(self, stream_state=None, **kwargs):
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        params["_type"] = self._type
+        if stream_state.get(self.cursor_field):
+            params["date_created__gt"] = stream_state.get(self.cursor_field)
+        return params
+
+
+class LeadCustomFields(CloseComStream):
+    def path(self, **kwargs) -> str:
+        return "custom_field/lead"
+
+
+class ContactCustomFields(CloseComStream):
+    def path(self, **kwargs) -> str:
+        return "custom_field/contact"
+
+
+class OpportunityCustomFields(CloseComStream):
+    def path(self, **kwargs) -> str:
+        return "custom_field/opportunity"
+
+
+class ActivityCustomFields(CloseComStream):
+    def path(self, **kwargs) -> str:
+        return "custom_field/activity"
+
+
+class Users(CloseComStream):
+    def path(self, **kwargs) -> str:
+        return "user"
+
+
+class SourceCloseCom(AbstractSource):
+    @staticmethod
+    def _convert_auth_to_token(username: str, password: str) -> str:
+        username = username.encode("latin1")
+        password = password.encode("latin1")
+        token = b64encode(b":".join((username, password))).strip().decode("ascii")
+        return token
+
+    def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, Any]:
+        try:
+            url = "https://api.close.com/api/v1/me"
+            response = requests.request(
+                "GET",
+                url=url,
+                auth=(config["api_key"], ""),
+            )
+            response.raise_for_status()
+            return True, None
+        except Exception as e:
+            return False, e
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        authenticator = TokenAuthenticator(auth_method="Basic", token=self._convert_auth_to_token(config["api_key"], ""))
```

Comment:
Do you need to use this custom method here?

Reply:
TokenAuthenticator should be used here